org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -92,7 +92,7 @@ private void importDir(Path rootFolder, IProgressMonitor monitor) {
 
 	public static GradleDistribution getGradleDistribution(Path rootFolder) {
 		GradleDistribution distribution = DEFAULT_DISTRIBUTION;
-		if (Files.exists(rootFolder.resolve("gradlew"))) {
+		if (Files.exists(rootFolder.resolve("gradle/wrapper/gradle-wrapper.properties"))) {
```

Comment:
you probably want to combine both checks, as the properties file could be present and the wrapper missing

Reply:
But the wrapper file is never actually used. Tooling API extracts the gradleDistUri property from the gradle-wrapper.properties and uses that directly.